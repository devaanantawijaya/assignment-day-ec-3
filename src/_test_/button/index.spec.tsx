import Button from "@/components/button";
import { render, fireEvent } from "@testing-library/react";

describe("Button", () => {
  it("should render the button and match snapshot", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Button title="Click" bg="bg-red-500" onClick={handleClick} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should trigger onClick when button is clicked", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button title="Click" bg="bg-blue-500" onClick={handleClick} />
    );
    const button = getByText("Click");
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("should render button with title", () => {
    const { getByText } = render(<Button title="Submit" bg="bg-yellow-500" />);
    expect(getByText("Submit")).toBeInTheDocument();
  });
});
