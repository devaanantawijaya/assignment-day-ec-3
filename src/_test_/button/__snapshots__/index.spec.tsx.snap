// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button and match snapshot 1`] = `
<div>
  <button
    class="px-5 py-2 rounded-xl w-full bg-red-500"
  >
    Click
  </button>
</div>
`;
